scriptoocloud/src/main/java/edu/asu/diging/scriptoocloud/core/service/impl/YamlParserService.java

```diff
@@ -0,0 +1,47 @@
+package edu.asu.diging.scriptoocloud.core.service.impl;
+
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.FileNotFoundException;
+import java.util.Map;
+
+import org.springframework.stereotype.Service;
+import org.yaml.snakeyaml.Yaml;
+
+@Service
+class YamlParserService  {
+
+    public Map<String,Object> parseYaml(String filePathString) throws FileNotFoundException{  
+        File file = new File(filePathString);
+        return parseYaml(file);
+    }
+
+    public Map<String,Object> parseYaml(File file) throws FileNotFoundException{  
```

Comment:
what is the logic for this method? Why does it iterate over a folder? what if there are several yaml files in a folder? Your parser should take one yaml file, or if there is other logic that is actually needed, which it looks like here, then the class should be named to something that describes that and the method as well.

Also, if the point of this class is to find the configuration file for the docker run, then it needs to look for the specific configuration yaml, not just any yaml (since there might be several yamls in a repository).